agenta-backend/agenta_backend/services/llm_apps_service.py

```diff
@@ -81,19 +81,22 @@ async def invoke_app(
     payload = await make_payload(datapoint, parameters, openapi_parameters)
 
     async with httpx.AsyncClient() as client:
-        logger.debug(f"Invoking app {uri} with payload {payload}")
-        response = await client.post(
-            url, json=payload, timeout=httpx.Timeout(timeout=5, read=None, write=5)
-        )
-        response.raise_for_status()
-
-        llm_app_response = response.json()
-        app_output = (
-            llm_app_response["message"]
-            if isinstance(llm_app_response, dict)
-            else llm_app_response
-        )
-        return AppOutput(output=app_output, status="success")
+        try:
+            logger.debug(f"Invoking app {uri} with payload {payload}")
+            response = await client.post(
+                url, json=payload, timeout=httpx.Timeout(timeout=5, read=None, write=5)
+            )
+            response.raise_for_status()
+
+            llm_app_response = response.json()
+            app_output = (
+                llm_app_response["message"]
+                if isinstance(llm_app_response, dict)
+                else llm_app_response
+            )
+            return AppOutput(output=app_output, status="success")
+        except:
+            return AppOutput(output="Error", status="error")
```

Comment:
The invoke_app function should only have one return statement. Please update this [line](https://github.com/Agenta-AI/agenta/pull/1270/files#diff-19f76e79daa870ae1cff9c52d9f2792ddc6ae0c4a4a09e3097d747fda14f74ffR138) with what you have here. Thank you.

Reply:
but it has only one return statement, no?

Reply:
What I mean is that there is no need to wrap it in a try-except because the run_with_retry takes care of the exception.

Reply:
That was not working previously. The reason I think is that we are not catching all exceptions. 
The current fix is just very short term. Akrem created a much better solution. And hopefully we would have that by the end of the day.
For now, if it's working let's move on with it